compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1665,6 +1665,7 @@ object desugar {
           Apply(sel, arg :: Nil).setApplyKind(ApplyKind.InfixTuple)
         case _ =>
           Apply(sel, arg :: Nil)
+      apply.withSpan(apply.span.union(arg.span))
```

Comment:
I tried
```
apply.withSpan(Span(left.span.start, right.span.end, point = op.span.start))
```
at the end of `binop`. That corrects some points in errors, and corresponds to "preserve the span". I'm not sure where point comes from in this expression. (Trying to make position calculations "read" correctly in Scala 2 was also a challenge.) Here `union` preserves point of the receiver, which is correct; I'll post the diffs.

Reply:
deceptively, `Span(start, end)` is not the same as `Span(start, end, point = start)` as I assumed for some reason. I have drilled down into synthetic spans for the "check unused" lint, but I really don't want to hold that arcana in my head. Positions should "just work", if we can figure out the best syntax.